prov/shm/src/smr_rma.c

```diff
@@ -152,27 +153,45 @@ ssize_t smr_generic_rma(struct smr_ep *ep, const struct iovec *iov,
 
 	total_len = ofi_total_iov_len(iov, iov_count);
 
-	if (total_len > SMR_INJECT_SIZE || op != ofi_op_write ||
-	    op_flags & FI_DELIVERY_COMPLETE) {
+	if (total_len < SMR_MSG_DATA_LEN && op != ofi_op_read_req &&
+			!(op_flags & FI_DELIVERY_COMPLETE)) {
+		smr_format_inline(cmd, smr_peer_addr(ep->region)[peer_id].addr,
+				  iov, iov_count, op, 0, data, op_flags);
+	} else if (total_len < SMR_INJECT_SIZE && op != ofi_op_read_req &&
+			!(op_flags & FI_DELIVERY_COMPLETE)) {
+		tx_buf = smr_freestack_pop(smr_inject_pool(peer_smr));
+		smr_format_inject(cmd, smr_peer_addr(ep->region)[peer_id].addr,
+				  iov, iov_count, op, 0, data, op_flags,
+				  peer_smr, tx_buf);
+	} else {
```

Comment:
This switching of the protocol order is unrelated to this patch. Please separate.

Reply:
Done.